README.md

```diff
@@ -113,22 +121,22 @@ A promise which resolves to the affine transform.
 ### `GDALDataset.convert(args)`
 Converts raster data between different formats. This is the equivalent of the [gdal_translate](https://gdal.org/programs/gdal_translate.html) command.
 
-**Note**: This returns a new `GDALDataset` object. It is necessary to call `.close()` on both the original dataset _and_ the new dataset in order to avoid memory leaks.
+**Note**: This returns a new `GDALDataset` object but does not perform any immediate calculation. Instead, calls to `.convert()` and `.warp()` are evaluated lazily. Each successive call to `.convert()` or `.warp()` is stored in a list of operations on the dataset object. These operations are only evaluated when necessary in order to access some property of the dataset, such as its size, bytes, or band count.
 #### Parameters
 - `args`: An array of strings, each representing a single command-line argument accepted by the `gdal_translate` command. The `src_dataset` and `dst_dataset` parameters should be omitted; these are handled by `GDALDataset`. Example: `ds.convert(['-outsize', '200%', '200%'])`
 #### Return value
-A promise that resolves to a new `GDALDataset` which has been converted according to the provided arguments to `gdal_translate`.
+A promise that resolves to a new `GDALDataset`.
 
 <br />
 
 ### `GDALDataset.warp(args)`
 Image reprojection and warping utility. This is the equivalent of the [gdalwarp](https://gdal.org/programs/gdalwarp.html) command.
 
-**Note**: This returns a new `GDALDataset` object. It is necessary to call `.close()` on both the original dataset _and_ the new dataset in order to avoid memory leaks.
+**Note**: This returns a new `GDALDataset` object but does not perform any immediate calculation. Instead, calls to `.convert()` and `.warp()` are evaluated lazily. Each successive call to `.convert()` or `.warp()` is stored in a list of operations on the dataset object. These operations are only evaluated when necessary in order to access some property of the dataset, such as its size, bytes, or band count.
```

Comment:
Yup pretty much. I originally implemented this feature using nested VRTs, defining a dataset as a list of VRT XML strings, which was nice because it was almost pure GDAL, only to discover that [VRTs hide overviews](https://gdal.org/drivers/raster/vrt.html#overviews) 🤦 . After that, I started kicking around the idea of doing it as a list of operation definitions, but didn't fully commit until I saw @pomadchin 's presentation on RasterSources and realized they're doing exactly the same thing, which convinced me that it was a workable approach.